datatypes/src/primitives/datetime.rs

```diff
@@ -0,0 +1,668 @@
+use crate::error::ErrorSource;
+use crate::primitives::TimeInstance;
+use chrono::{Datelike, NaiveDate, NaiveDateTime, Offset, Timelike};
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
+use std::cmp::Ordering;
+use std::ops::{Add, Sub};
+use std::str::FromStr;
+
+/// An object that composes the date and a timestamp with time zone.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct DateTime {
+    datetime: NaiveDateTime,
+    offset: UtcOffset,
```

Comment:
internally we always work with UTC datetimes/TimeInstance. The only point where we need to handle timezones is during parsing/serializing. I think it would make sense to have separate types. One DateTime type for the whole TimeStep stuff that is always UTC and one DateTime type with timezones for in/output to/from the user

Reply:
We could use it for computations like: Give me all points from noon wherever your noon is at.
Because in UTC time, it is like. Well, for you it is noon, but we treat it being in the middle of the night.

I guess the other problem is, that Postgres does not store it this way.
So we should discuss the use case I guess briefly in the team.
Currently, we have just ignored it and used the library as is. But now it could be of interest :shrug:

Reply:
[9b18fc7](https://github.com/geo-engine/geoengine/pull/523/commits/9b18fc7ba1c2a7fc0fbe2882cec07a2f7a2b1108)

[003ff00](https://github.com/geo-engine/geoengine/pull/523/commits/003ff00ac566c68568deb8c4c306f29e3646ad46)